packages/yourloops/lib/auth/api.ts

```diff
@@ -684,28 +684,78 @@ class AuthApi extends EventTarget {
     let matomoPaq = null;
     this.log.info("Metrics:", eventName, properties);
     switch (appConfig.METRICS_SERVICE) {
-    case "matomo":
-      matomoPaq = window._paq;
-      if (!_.isObject(matomoPaq)) {
-        this.log.error("Matomo do not seems to be available, wrong configuration");
-        return;
-      }
-      if (eventName === "CookieConsent") {
-        matomoPaq.push(["setConsentGiven", properties]);
-      } else if (eventName === "setCustomUrl") {
-        matomoPaq.push(["setCustomUrl", properties]);
-      } else if (eventName === "setUserId") {
-        matomoPaq.push(["setUserId", properties]);
-      } else if (eventName === "resetUserId") {
-        matomoPaq.push(["resetUserId"]);
-      } else if (eventName === "setDocumentTitle" && typeof properties === "string") {
-        matomoPaq.push(["setDocumentTitle", properties]);
-      } else if (typeof properties === "undefined") {
-        matomoPaq.push(["trackEvent", eventName]);
-      } else {
-        matomoPaq.push(["trackEvent", eventName, JSON.stringify(properties)]);
-      }
-      break;
+      case "matomo":
+        matomoPaq = window._paq;
+        if (!_.isObject(matomoPaq)) {
+          this.log.error("Matomo do not seems to be available, wrong configuration");
+          return;
+        }
+        if (eventName === "CookieConsent") {
+          matomoPaq.push(["setConsentGiven", properties]);
+        } else if (eventName === "setCustomUrl") {
+          matomoPaq.push(["setCustomUrl", properties]);
+        } else if (eventName === "setUserId") {
+          matomoPaq.push(["setUserId", properties]);
+        } else if (eventName === "resetUserId") {
+          matomoPaq.push(["resetUserId"]);
+        } else if (eventName === "setDocumentTitle" && typeof properties === "string") {
+          matomoPaq.push(["setDocumentTitle", properties]);
+        } else if (typeof properties === "undefined") {
+          matomoPaq.push(["trackEvent", eventName]);
+        } else {
+          matomoPaq.push(["trackEvent", eventName, JSON.stringify(properties)]);
+        }
+        break;
+    }
+  }
+
+  public async updateUserProfile({ userid, profile }: User): Promise<void> {
+    if (!this.isLoggedIn) {
+      // Users should never see this:
+      throw new Error(t("You are not logged-in"));
+    }
+
+    const seagullURL = new URL(`/metadata/${userid}/profile`, appConfig.API_HOST);
+
+    const response = await fetch(seagullURL.toString(), {
+      method: "POST",
+      headers: {
```

Comment:
Missing Content-Type 
```suggestion
      headers: {
        "Content-Type": "application/json",
```